src/transform/FirstParagraphRelocationTransform.js

```diff
@@ -0,0 +1,92 @@
+import Polyfill from './Polyfill'
+
+/**
+ * Determine if paragraph is the one we are interested in.
+ * @param  {!HTMLParagraphElement}  paragraphElement
+ * @return {boolean}
+ */
+const isParagraphGood = paragraphElement => {
+  // Ignore 'coordinates' which are presently hidden. See enwiki 'Bolton Field' and 'Sharya Forest
+  // Museum Railway'. Not counting coordinates towards the min 'good' textContent length heuristic
+  // has dual effect of p's containing only coordinates being rejected, and p's containing
+  // coordinates but also other elements meeting the min 'good' textContent length being accepted.
+  const coordElement = paragraphElement.querySelector('[id="coordinates"]')
+  const coordTextLength = !coordElement ? 0 : coordElement.textContent.length
+
+  // Ensures the paragraph has at least a little text. Otherwise silly things like a empty P or P
+  // which only contains a BR tag will get pulled up. See enwiki 'Hawaii', 'United States',
+  // 'Academy (educational institution)', 'Lovászpatona'
+  const minLength = 50
+  const lengthExceedsMinLength =
+    paragraphElement.textContent.length - coordTextLength >= minLength
+  return lengthExceedsMinLength
+}
+
+/**
+ * Everything we want to move up (everything between `goodParagraphElement` and the next paragraph).
+ * @param  {!HTMLParagraphElement} goodParagraphElement
+ * @return {!Array.<Element>}
+ */
+const getElementsToMove = goodParagraphElement => {
+  let didHitP = false
+  let didHitNextP = false
+  const shouldElementMoveUp = element => { // eslint-disable-line require-jsdoc
+    if (didHitP && element.tagName === 'P') {
+      didHitNextP = true
+    } else if (element.isEqualNode(goodParagraphElement)) {
+      didHitP = true
+    }
+    return didHitP && !didHitNextP
+  }
+  return Array.from(goodParagraphElement.parentNode.childNodes).filter(shouldElementMoveUp)
```

Comment:
Babel said it supports `Array.from()`... but ya I can tweak it if needed.

I'll check on 2 and 3...

Reply:
We don't include babel-polyfill though, only transpilation. Looking at the Git log, I see a number of cases where we've retroactively removed Array.from() usage. I don't know if the targets we support still need a polyfill or not but something to keep in mind.

https://babeljs.io/docs/usage/caveats/#polyfills

Reply:
Ah thanks for the reminder!!! :)

Reply:
1 is fixed. Looking at 2 and 3 now...

Reply:
Regarding using general sibling selector on the parent, it's a bit complicated because it's hard to encode the `goodness` heuristic in a query selector. That's not a show stopper because you can mark the good element by adding an attribute and using `[attribute=value]`, but there's one more issue - it's not apparent to me how you'd make a general sibling selector halt at the next paragraph it encounters. We want to grab the `good` P and everything up to but not including the next sibling P. You can do `p[isGoodStuff=true] ~ *:not(p)`, but that doesn't halt at the next P - it just excludes them and keeps matching other elements...

Reply:
This https://stackoverflow.com/a/42592920/135557 seems to confirm my suspicions about CSS level 3 selectors not quite being capable of handling what we're trying to do...

Reply:
I renamed `getElementsToMove` to `getNodesToMove` to make clear that things like text nodes can be moved too. Also added a test that text nodes which are expected to be moved are moved..

Reply:
Trying to re-work it now to use `nextSibling`... looks like that would simplify it a bit :)

Reply:
@niedzielski I simplified it using nextSibling. Nice because it doesn't need to maintain as much state.